File: fixtures/scale/ruby/large/lib/check_skewmap.rb
require_relative 'skewmap_encrypt'
require_relative 'skewmap_decrypt'
require_relative 'corpus_affine'

def check_skewmap
  corpus_affine.each do |text|
    sealed = skewmap_encrypt(text, 16)
    opened = skewmap_decrypt(sealed, 16)
    return false if opened != text
  end
  true
end

File: fixtures/scale/python/large/cipherlab/checks/check_skewmap.py
from cipherlab.ciphers.skewmap import skewmap_encrypt, skewmap_decrypt
from cipherlab.specs.skewmap_spec import skewmap_spec
from cipherlab.corpus.corpus_affine import corpus_affine


def check_skewmap():
    spec = skewmap_spec()
    for text in corpus_affine():
        sealed = skewmap_encrypt(text, spec["key"])
        opened = skewmap_decrypt(sealed, spec["key"])
        if opened != text:
            return False
    return spec["category"] == "affine"
